### PR TITLE
LUCENE-10524 Add benchmark suite details to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by gautamworah96**
Today, new contributors are usually unaware of where luceneutil benchmarks are and when/how to run them. Committers usually end up pointing contributors to the benchmarks package when they make perf impacting changes and then they run the benchmarks.

Adding benchmark details to the Lucene repo will also make them more accessible to other researchers who want to experiment/benchmark their own custom task implementation with Java Lucene.

JIRA: https://issues.apache.org/jira/browse/LUCENE-10524

cc: @mocobeta

### CODE REVIEW

**Comment by mocobeta** [comment]
@gautamworah96 This looks good to me!
I wonder if it makes sense to mention the off-the-shelf https://github.com/apache/lucene/tree/main/lucene/benchmark module here too (I didn't find instruction for running it, but a mention still could be helpful).

I'd also suggest letting @mikemccand know the change, thanks.

**Review by mocobeta — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,6 +66,16 @@ In case your contribution fixes a bug, please create a new test case that fails
 - *Eclipse*  - Basic support ([help/IDEs.txt](https://github.com/apache/lucene/blob/main/help/IDEs.txt#L7)).
 - *Netbeans* - Not tested.
 
+## Benchmarking 
+
+Use the tool suite at [luceneutil](https://github.com/mikemccand/luceneutil) to benchmark your code changes
+if you think that your change may have measurably changed the performance of a task.
```
Comment (mikemccand) [inline CONTRIBUTING.md:72]: Maybe add "These are the same benchmarks that run in [nightly benchmarks](https://home.apache.org/~mikemccand/lucenebench/)" or so?

**Review by mikemccand — APPROVED** [review]
Thanks @gautamworah96 and @mocobeta!  This looks great.

> I wonder if it makes sense to mention the off-the-shelf https://github.com/apache/lucene/tree/main/lucene/benchmark module here too (I didn't find instruction for running it, but a mention still could be helpful).
 
+1 to also make a quick mention of the `benchmark` module as well.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -66,6 +66,16 @@ In case your contribution fixes a bug, please create a new test case that fails
...
+
+The instructions for running the benchmarks can be found in the luceneutil [README](https://github.com/mikemccand/luceneutil/blob/master/README.md).
+
+The Lucene community is also interested in other implementations of these benchmark tasks.
+Feel free to share your findings (especially if your implementation performs better!) through the [Lucene mailing lists](https://lucene.apache.org/core/discussion.html).
```
[the remark this answers is not in the document]
Comment (gautamworah96) [reply CONTRIBUTING.md]: Added

File: CONTRIBUTING.md (hunk 1 above)
Comment (gautamworah96) [reply CONTRIBUTING.md]: Added

**Review by mocobeta — APPROVED** [review]
@gautamworah96 thanks for looking at this!

**Pull request merged**